src/Uno.UI.Composition/Composition/Visual.skia.cs

```diff
@@ -177,23 +184,60 @@ internal virtual void Render(in DrawingSession parentSession)
 			return;
 		}
 
-		using var session = BeginDrawing(in parentSession);
-		Draw(in session);
+		using var session = parentSession.WithOpacity(Opacity);
+		SetupLocalCoordinatesAndClip(in session);
+		Paint(in session);
+
+		// The CornerRadiusClip doesn't affect the visual itself, only its children
+		CornerRadiusClip?.Apply(session.Canvas, this);
+
+		foreach (var child in GetChildrenInRenderOrder())
+		{
+			child.Render(in session);
+		}
```

Comment:
Not sure I'm a fan of this ... sounds like if we are again trying to support all cases in the `Visual` instead of deferring the responsibility in each sub-class.